*** makefile.bcc	2007-09-28 13:59:22.711122363 +0100
--- makefile.bcc.patched	2007-09-28 13:59:49.269529705 +0100
***************
*** 19,29 ****
  !endif
  
  # The name of your C compiler:
! CC= bcc
  
  # You may need to adjust these cc options:
  !if $d(DOS)
! CFLAGS= -O2 -mm -w-par -w-stu -w-ccc -w-rch
  !else
  CFLAGS= -O1 -w-par -w-stu -w-ccc -w-rch
  !endif
--- 19,29 ----
  !endif
  
  # The name of your C compiler:
! CC= bcc32
  
  # You may need to adjust these cc options:
  !if $d(DOS)
! CFLAGS= -O2 -w-par -w-stu -w-ccc -w-rch
  !else
  CFLAGS= -O1 -w-par -w-stu -w-ccc -w-rch
  !endif
***************
*** 40,46 ****
  
  # Link-time cc options:
  !if $d(DOS)
! LDFLAGS= -mm
  # memory model option here must match CFLAGS!
  !else
  LDFLAGS=
--- 40,46 ----
  
  # Link-time cc options:
  !if $d(DOS)
! LDFLAGS=
  # memory model option here must match CFLAGS!
  !else
  LDFLAGS=
***************
*** 54,61 ****
  # For OS/2, we recommend jmemnobs.c (flat memory!)
  # SYSDEPMEMLIB must list the same files with "+" signs for the librarian.
  !if $d(DOS)
! SYSDEPMEM= jmemdos.obj jmemdosa.obj
! SYSDEPMEMLIB= +jmemdos.obj +jmemdosa.obj
  !else
  SYSDEPMEM= jmemnobs.obj
  SYSDEPMEMLIB= +jmemnobs.obj
--- 54,61 ----
  # For OS/2, we recommend jmemnobs.c (flat memory!)
  # SYSDEPMEMLIB must list the same files with "+" signs for the librarian.
  !if $d(DOS)
! SYSDEPMEM= jmemsys.obj
! SYSDEPMEMLIB= +jmemsys.obj
  !else
  SYSDEPMEM= jmemnobs.obj
  SYSDEPMEMLIB= +jmemnobs.obj
***************
*** 152,158 ****
  
  rdjpgcom.exe: rdjpgcom.c
  !if $d(DOS)
! 	$(CC) -ms -O rdjpgcom.c
  !else
  	$(CC) $(CFLAGS) rdjpgcom.c
  !endif
--- 152,158 ----
  
  rdjpgcom.exe: rdjpgcom.c
  !if $d(DOS)
! 	$(CC) -O rdjpgcom.c
  !else
  	$(CC) $(CFLAGS) rdjpgcom.c
  !endif
***************
*** 160,166 ****
  # On DOS, wrjpgcom needs large model so it can malloc a 64K chunk
  wrjpgcom.exe: wrjpgcom.c
  !if $d(DOS)
! 	$(CC) -ml -O wrjpgcom.c
  !else
  	$(CC) $(CFLAGS) wrjpgcom.c
  !endif
--- 160,166 ----
  # On DOS, wrjpgcom needs large model so it can malloc a 64K chunk
  wrjpgcom.exe: wrjpgcom.c
  !if $d(DOS)
! 	$(CC) -O wrjpgcom.c
  !else
  	$(CC) $(CFLAGS) wrjpgcom.c
  !endif
