# use patch -p0 < filename
# created with diff -crB
*** Coin-3.1.3/src/nodes/SoSeparator.cpp    2009-01-14 16:26:08.000000000 +0000
--- ./Coin-3.1.3/src/nodes/SoSeparator.cpp  2009-01-14 16:26:08.000000000 +0000
***************
*** 771,776 ****
--- 771,787 ----
            action->popCurPath(pathcode);
          }
        }
+ 
+       // fix by Zsolt Husz, HGU, MRC 
+       int maxind=-1;
+       {
+ 	for (int i = 0; i < numindices; i++)
+ 	if (maxind<indices[i])  maxind = indices[i];
+       }
+       if (childidx>maxind) {
+ 	childidx=indices[numindices-1];
+       }
+       
        SoNode * inpath = childarray[childidx];
        action->pushCurPath(childidx, inpath);
        if (!action->abortNow()) {
